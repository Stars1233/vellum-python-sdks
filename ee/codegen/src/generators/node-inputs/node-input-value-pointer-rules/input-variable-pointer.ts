import { python } from "@fern-api/python-ast";

import { BaseNodeInputValuePointerRule } from "./base";

import { InputVariablePointer } from "src/types/vellum";
import { toSnakeCase } from "src/utils/casing";

export class InputVariablePointerRule extends BaseNodeInputValuePointerRule<InputVariablePointer> {
  getAstNode(): python.Reference {
    const inputVariablePointerRuleData = this.nodeInputValuePointerRule.data;

    const inputVariableContext =
      this.workflowContext.getInputVariableContextById(
        inputVariablePointerRuleData.inputVariableId
      );

    return python.reference({
      name: "Inputs",
      modulePath: inputVariableContext.modulePath,
      attribute: [toSnakeCase(inputVariableContext.getInputVariableName())],
    });
  }
}
