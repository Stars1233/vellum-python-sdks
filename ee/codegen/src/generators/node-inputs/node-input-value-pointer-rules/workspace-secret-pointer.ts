import { python } from "@fern-api/python-ast";

import { BaseNodeInputValuePointerRule } from "./base";

import { WorkspaceSecretPointer as WorkspaceSecretPointerType } from "src/types/vellum";

export class WorkspaceSecretPointerRule extends BaseNodeInputValuePointerRule<WorkspaceSecretPointerType> {
  getAstNode(): python.AstNode {
    const workspaceSecretPointerData = this.nodeInputValuePointerRule.data;

    const workspaceSecretName = workspaceSecretPointerData.workspaceSecretId;

    if (!workspaceSecretName) {
      throw new Error("Workspace secret name is required");
    }

    return python.instantiateClass({
      classReference: python.reference({
        name: "VellumSecretReference",
        modulePath: [
          ...this.workflowContext.sdkModulePathNames.WORKFLOWS_MODULE_PATH,
          "references",
        ],
      }),
      arguments_: [
        python.methodArgument({
          value: python.TypeInstantiation.str(workspaceSecretName),
        }),
      ],
    });
  }
}
