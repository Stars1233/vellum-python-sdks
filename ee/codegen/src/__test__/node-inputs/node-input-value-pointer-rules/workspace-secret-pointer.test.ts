import { Writer } from "@fern-api/python-ast/core/Writer";

import { workflowContextFactory } from "src/__test__/helpers";
import { WorkspaceSecretPointerRule } from "src/generators/node-inputs/node-input-value-pointer-rules/workspace-secret-pointer";

describe("WorkspaceSecretPointer", () => {
  let writer: Writer;

  beforeEach(() => {
    writer = new Writer();
  });

  afterEach(() => {
    vi.restoreAllMocks();
  });

  it("should generate correct Python code", async () => {
    const workflowContext = workflowContextFactory();

    const workspaceSecretPointer = new WorkspaceSecretPointerRule({
      workflowContext: workflowContext,
      nodeInputValuePointerRule: {
        type: "WORKSPACE_SECRET",
        data: {
          type: "STRING",
          workspaceSecretId: "MY_SECRET",
        },
      },
    });

    workspaceSecretPointer.write(writer);
    expect(await writer.toStringFormatted()).toMatchSnapshot();
  });
});
