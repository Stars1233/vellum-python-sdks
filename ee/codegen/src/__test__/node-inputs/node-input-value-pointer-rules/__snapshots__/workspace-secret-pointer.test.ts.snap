// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WorkspaceSecretPointer > should generate correct Python code 1`] = `
"VellumSecretReference("MY_SECRET")
"
`;
